File: de.lmu.ifi.dbs.knowing.core.swt.charts/src/de/lmu/ifi/dbs/knowing/core/swt/charts/internal/Activator.java
package de.lmu.ifi.dbs.knowing.core.swt.charts.internal;

import org.osgi.framework.BundleActivator;
import org.osgi.framework.BundleContext;

import de.lmu.ifi.dbs.knowing.core.swt.charts.PiePresenterFactory;
import de.lmu.ifi.dbs.knowing.core.swt.charts.TimePeriodValuesPresenter;
import de.lmu.ifi.dbs.knowing.core.util.FactoryUtil;

public class Activator implements BundleActivator {

	private static BundleContext context;

	static BundleContext getContext() {
		return context;
	}

	/*
	 * (non-Javadoc)
	 * @see org.osgi.framework.BundleActivator#start(org.osgi.framework.BundleContext)
	 */
	public void start(BundleContext bundleContext) throws Exception {
		Activator.context = bundleContext;
		
		//Registering all factories
		FactoryUtil.registerPresenterFactory(new PiePresenterFactory(), null, bundleContext);
		FactoryUtil.registerPresenterFactory(new TimePeriodValuesPresenter(), null, bundleContext);
	}

	/*
	 * (non-Javadoc)
	 * @see org.osgi.framework.BundleActivator#stop(org.osgi.framework.BundleContext)
	 */
	public void stop(BundleContext bundleContext) throws Exception {
		Activator.context = null;
	}


}

File: de.lmu.ifi.dbs.knowing.core.swt.charts/src/de/lmu/ifi/dbs/knowing/core/swt/charts/internal/Activator.scala
package de.lmu.ifi.dbs.knowing.core.swt.charts.internal

import de.lmu.ifi.dbs.knowing.core.swt.charts._
import de.lmu.ifi.dbs.knowing.core.factory.TFactory
import de.lmu.ifi.dbs.knowing.core.util.OSGIUtil
import org.osgi.framework.{ BundleContext, BundleActivator, ServiceRegistration }

/**
 * @author Nepomuk Seiler
 * @version 0.1
 * @since 25.04.2011
 *
 */
class Activator extends BundleActivator {

	private var osgi: OSGIUtil = _

	def start(context: BundleContext) = {
		osgi = new OSGIUtil(context)
		osgi.registerPresenter(new PieChartPresenterFactory)
		osgi.registerPresenter(new TimeIntervalClassPresenterFactory)
		osgi.registerPresenter(new TimeSeriesPresenterFactory)
		osgi.registerPresenter(new AreaChartPresenterFactory)
	}

	def stop(context: BundleContext) = {
		osgi.deregisterAll
		osgi = null
	}

}